resources/prosody-plugins/mod_muc_system_chat_message.lua

```diff
@@ -0,0 +1,140 @@
+-- A global module which can be used as http endpoint to send system chat messages to meeting participants. The provided token
+--- in the request is verified whether it has the right to do so.
+-- Copyright (C) 2024-present 8x8, Inc.
+
+-- curl https://{host}/send-system-message  -d '{"message": "testmessage", "to": "{connection_jid}", "room": "{room_jid}"}' -H "content-type: application/json" -H "authorization: Bearer {token}"
+
+module:set_global();
+
+local util = module:require "util";
+
+local async_handler_wrapper = util.async_handler_wrapper;
+local room_jid_match_rewrite = util.room_jid_match_rewrite;
+local starts_with = util.starts_with;
+local get_room_from_jid = util.get_room_from_jid;
+
+local st = require "util.stanza";
+local json = require "util.json";
+local inspect = require 'inspect';
+
+-- will be initialized once the main virtual host module is initialized
+local token_util;
+
+local muc_domain_base = module:get_option_string("muc_mapper_domain_base");
+local asapKeyServer = module:get_option_string("prosody_password_public_key_repo_url", "");
+
+function verify_token(token)
+    if token == nil then
+        module:log("warn", "no token provided");
+        return false;
+    end
+
+    local session = {};
+    session.auth_token = token;
+    local verified, reason, msg = token_util:process_and_verify_token(session);
+    if not verified then
+        module:log("warn", "not a valid token %s %s", tostring(reason), tostring(msg));
+        return false;
+    end
+    return true;
+end
+
+function handle_send_system_message (event)
+    local request = event.request;
+
+    module:log("info", "Request for sending a system message received: reqid %s", request.headers["request_id"])
```

Comment:
Will this be used a lot? Will this be chatty? 
Can you make it debug, or at least move it and print in which room was that used, otherwise this log means nothing when debugging what happens in a room.

Reply:
ill mark it as debug, since this was the original purpose

Reply:
Look at the other modules and how debug prints are behind a flag, that is for performance purposes. 
As strings and replacements are done even when not printed, although its not on a hot path its good to have code consistent :) Thanks.